File: src/main/java/com/EmosewaPixel/pixellib/materialSystem/materials/HarvestTier.java
package com.EmosewaPixel.pixellib.materialSystem.materials;

public class HarvestTier {
    private float hardness;
    private float resistance;
    private int level;

    public HarvestTier(float hardness, float resistance, int level) {
        this.hardness = hardness;
        this.resistance = resistance;
        this.level = level;
    }

    public float getHardness() {
        return hardness;
    }

    public float getResistance() {
        return resistance;
    }

    public int getHarvestLevel() {
        return level;
    }
}

File: src/main/java/com/emosewapixel/pixellib/materialsystem/materials/HarvestTier.kt
package com.emosewapixel.pixellib.materialsystem.materials

import com.blamejared.crafttweaker.api.annotations.ZenRegister
import org.openzen.zencode.java.ZenCodeType

//Harvest Tiers contain the stats for breaking blocks
@ZenRegister
@ZenCodeType.Name("pixellib.materialsystem.materials.HarvestTier")
data class HarvestTier @ZenCodeType.Constructor constructor(
        @ZenCodeType.Field val hardness: Float,
        @ZenCodeType.Field val resistance: Float,
        @ZenCodeType.Field val harvestLevel: Int
)